libdnf/hy-subject.c

```diff
@@ -216,7 +219,171 @@ hy_possibilities_next_nevra(HyPossibilities iter, HyNevra *out_nevra)
                 return 0;
         }
         form = iter->forms[iter->current];
-        hy_nevra_free(*out_nevra);
+        g_clear_pointer(out_nevra, hy_nevra_free);
     }
     return -1;
 }
+
+struct NevraToQuery {
+    int nevra_type;
+    int query_type;
+};
+
+/* Given a subject, attempt to create a query choose the first one, and update
+ * the query to try to match it.
+ * 
+ * This code is based on rpm-software-management/dnf/subject.py:get_best_query() at git
+ * revision: 1d83fdc0280ca4202281ef489afe600e2f51a32a
+ */
+HyQuery
+hy_subject_get_best_query(HySubject subject, DnfSack *sack, gboolean with_provides)
+{
+#if 0
```

Comment:
rpm-ostree and PK does not (and does not want to) support `multilib_policy` config option? I also think that this should be supported (not necessary in this PR though).